File: vs_xam/baby-flashcards/ViewLogic/DeckViewCell.cs
using System;
using UIKit;
using CoreGraphics;
using Foundation;
using SpriteKit;
using CoreAnimation;

namespace babyflashcards
{
	public class DeckViewCell : UICollectionViewCell
	{
		private readonly UIImageView _imageView;

		[Export ("initWithFrame:")]
		public DeckViewCell (CGRect frame) : base (frame)
		{
			/*			ContentView.Layer.BorderColor = UIColor.LightGray.CGColor;
						ContentView.Layer.CornerRadius = 5;
						ContentView.Layer.MasksToBounds = true;
						ContentView.Layer.BorderWidth = 2.0f;
			ContentView.Transform = CGAffineTransform.MakeScale (0.8f, 0.8f); */

			ContentView.BackgroundColor = UIColor.Clear;
			ContentView.Opaque = false;

			_imageView = new UIImageView (frame);
			_imageView.ContentMode = UIViewContentMode.ScaleAspectFit;
			_imageView.ClipsToBounds = true;
			_imageView.Layer.MasksToBounds = true;
			_imageView.Center = ContentView.Center;
			_imageView.Opaque = false;

			//imageView.Transform = CGAffineTransform.MakeScale (0.7f, 0.7f);

			ContentView.AddSubview (_imageView);
		}

		public override CALayer Layer {
			get {
				base.Layer.Speed = 0.4f;
				return base.Layer;
			}
		}

		public String ImagePath {
			set {
				_imageView.Image = UIImage.FromBundle (value);;
			}
		}

	}
}

File: xcode/Scratch1/Scratch1/DeckViewCell.swift
//
//  DeckViewCell.swift
//  baby-flashcards
//
//  Created by Michael Ishmael on 16/05/2016.
//  Copyright © 2016 66Bytes. All rights reserved.
//

import Foundation
import UIKit

class DeckViewCell : UICollectionViewCell {
    
    private var _imageView:UIImageView
    
    override init(frame:CGRect){
        _imageView = UIImageView(frame: frame);
        super.init(frame: frame)
        self.contentView.backgroundColor = UIColor.clearColor();
        self.contentView.opaque = false;
        
        
        _imageView.contentMode = UIViewContentMode.ScaleToFill;
        _imageView.clipsToBounds = true;
        _imageView.layer.masksToBounds = true;
        _imageView.center = self.contentView.center;
        _imageView.opaque = false;
    
        //_imageView.transform = CGAffineTransformMakeScale(0.8, 0.8)
        //CGAffineTransform.MakeScale (0.7f, 0.7,f);
        
        self.contentView.addSubview(_imageView);
        
    }

    
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override internal var layer:CALayer {
        get{
            super.layer.speed = 0.4;
            return super.layer
        }
    }
    
    func setImagePath(value:String) {
        
        _imageView.image = UIImage.init(named: value)
        
    }
    
//    public override CALayer Layer {
//    get {
//				base.Layer.Speed = 0.4f;
//				return base.Layer;
//    }
//    }
//    
//    public String ImagePath {
//    set {
//				_imageView.Image = UIImage.FromBundle (value);;
//    }
//    }
    
    
    
}